sap-spring-properties/src/main/java/com/sap/cloud/environment/servicebinding/SapSpringPropertiesServiceBindingAccessor.java

```diff
@@ -0,0 +1,97 @@
+package com.sap.cloud.environment.servicebinding;
+
+import com.sap.cloud.environment.servicebinding.api.DefaultServiceBinding;
+import com.sap.cloud.environment.servicebinding.api.ServiceBinding;
+import com.sap.cloud.environment.servicebinding.api.ServiceBindingAccessor;
+import com.sap.cloud.environment.servicebinding.api.exception.ServiceBindingAccessException;
+import com.sap.cloud.environment.servicebinding.environment.SapServiceBindingsPropertiesAccessor;
+import com.sap.cloud.environment.servicebinding.SapServiceBindingsProperties.ServiceBindingProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * A {@link ServiceBindingAccessor} that is able to load {@link SapServiceBindingsProperties}s from Spring's application
+ * properties.
+ */
+public class SapSpringPropertiesServiceBindingAccessor implements ServiceBindingAccessor
+{
+    @Nonnull
+    private static final Logger logger = LoggerFactory.getLogger(SapSpringPropertiesServiceBindingAccessor.class);
+
+    @Nonnull
+    @Override
+    public List<ServiceBinding> getServiceBindings()
+        throws ServiceBindingAccessException
+    {
+        List<ServiceBinding> serviceBindings = fetchServiceBindings();
+
+        serviceBindings
+            .forEach(binding -> logger.debug("Service binding {} found in properties.", binding.getServiceName().get()) //NOSONAR
+            );
+
+        return serviceBindings;
+    }
+
+    @Nonnull
+    private List<ServiceBinding> fetchServiceBindings()
+    {
+        Map<String, ServiceBindingProperties> serviceBindingsProperties =
+            SapServiceBindingsPropertiesAccessor.getServiceBindingsProperties();
+
+        if( serviceBindingsProperties.isEmpty() ) {
+            logger.info("No service bindings found in properties.");
+        }
+
+        return serviceBindingsProperties
+            .entrySet()
+            .stream()
+            .map(entry -> toServiceBinding(entry.getKey(), entry.getValue()))
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+    }
+
+    @Nullable
+    private ServiceBinding toServiceBinding(
+        @Nonnull String serviceBindingName,
+        @Nonnull final ServiceBindingProperties serviceBindingProperties )
+    {
+        if( validateServiceBindingProperties(serviceBindingProperties) ) {
+            return DefaultServiceBinding
+                .builder()
+                .copy(Collections.emptyMap())
+                .withName(serviceBindingName)
```

Comment:
If `serviceBindingProperties.getName()` is set I think it should override using the properties key `serviceBindingName`.

Reply:
yes, but why should one duplicate that information (property-name and then again at a the property "name")? I just removed the unnecessary property "name".

Reply:
We typically use that pattern in CAP for cases where the desired name can't be easily expressed as a YAML key. For example Spring Boot doesn't like `_` in YAML keys and you need to quote that with an ugly `'[MY_KEY_WITH_UNDERSCORES]'`. In general there are less restrictions on YAML keys than on YAML values. Therefore I'd suggest to keep it, but default to the YAML key in case the name is not explicitly given. Then it is a totally optional property, but provides flexibility in certain edge case scenarios.